scipy/stats/tests/test_generation/reference_distribution_infrastructure_tests.py

```diff
@@ -0,0 +1,41 @@
+# Note: this file is to be run locally, not on CI. It is only for tests of the
+# reference distribution *infrastructure*; unit tests of SciPy distributions
+# do not go here.
+
+import numpy as np
+from scipy import stats
+from numpy.testing import assert_allclose
+import scipy.stats.tests.test_generation.reference_distributions as rd
+from mpmath import mp
+
+def test_basic():
+    mp.dps = 20  # high enough to pass, not unreasonably slow
```

Comment:
Meaning bellow 1s or more?

Reply:
ok let's mark it slow then.

(Note: we somehow miss a marker for things which could just be run on a single job and not all over the place.)

Reply:
I don't plan on this being run with the regular test suite. I don't think we want to add an `mpmath` dependency, so the filename does not begin with `test` and is therefore excluded.

If a bug is introduced in this code later, it will start generating inaccurate reference values. Presumably, the values produced by the SciPy distribution under development will disagree with it. I think this file will be valuable at that point. The developer can run this test manually to see if the problem is with the infrastructure or the distribution.

I think the more likely scenario is that this code will produce inaccurate values even without a bug. We cannot pretend the mpmath is infallible. I've seen it produce wildly inaccurate results even with extremely high `dps` - and that was just evaluating some standard functions (at extreme values), not even using `quad` or `findroot`.

To guard against this, it would be good practice to run the results at multiple `dps` settings and see if the results agree exactly to 16 digits, but probably *not* in all digits generated. We can add some code to automate this and generate plots that show convergence as `dps` is increased. That would be a follow-up.

Reply:
Ok fine by me, thanks for the detailed explanation 👍 (and oops I missed it was not actually being run.)